### PR TITLE
[tex] Fix for search_env_filtered(), (bnc#934227)

### PR DESCRIPTION
**Pull request by mjura**
Check if node[barclamp][:config] exist to prevent

    undefined method `[]' for nil:NilClass

(cherry picked from commit 763ff24)

This is needed if want to merge https://github.com/crowbar/barclamp-neutron/pull/223

### CODE REVIEW

**Comment by bkutil** [comment]
+1

File: chef/cookbooks/utils/libraries/search.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def search_env_filtered(type, query="*:*", sort='X_CHEF_id_CHEF_X asc',
       # There are two conventions to filter by barclamp proposal:
       #  1) Other barclamp cookbook: node[@cookbook_name][$OTHER_BC_NAME_instance]
       #  2) Same cookbook: node[@cookbook_name][:config][:environment]
-      if barclamp == cookbook_name
+      if !node[barclamp].nil? && !node[barclamp][:config].nil? && (barclamp == cookbook_name)
```
Comment (tboerger) [inline chef/cookbooks/utils/libraries/search.rb]: As mentioned in the other pull request:

```
if node[barclamp] &&!node[barclamp][:config] && (barclamp == cookbook_name)
```

Comment (mjura) [inline chef/cookbooks/utils/libraries/search.rb]: Done

**Comment by bkutil** [comment]
Still +1 ;)

**Comment by tboerger** [comment]
Ah, you can skip the ```()```, otherwise :+1:

**Comment by dirkmueller** [comment]
-1, the git commit message is referring the wrong commit (cherry-pick -x again please)

**Comment by mjura** [comment]
> -1, the git commit message is referring the wrong commit (cherry-pick -x again please)

@dirkmueller : thank you for seeing this, I fixed the comment

**Pull request merged**